aten/src/ATen/native/mkldnn/Linear.cpp

```diff
@@ -28,16 +28,22 @@ Tensor mkldnn_linear(
     const Tensor& self,
     const Tensor& weight,
     const Tensor& bias) {
-  TORCH_CHECK(self.dim() >= 2,
-      "mkldnn_linear: input needs to has dim at least 2, input dim ", self.dim());
+  const int64_t dim = self.dim();
+  TORCH_CHECK(
+      self.dim() != 0,
```

Comment:
I wonder if there was a performance concern here. 
MKLDNN also has `inner_product`: https://oneapi-src.github.io/oneDNN/structmkldnn_1_1inner__product__forward_1_1desc.html#details We could ask Intel guys.

Reply:
Yea, that's a good idea. With dim > 2 they just do a reshape, so this is equivalent to that approach. From looking at the top 200 shapes internally broadcasting linear is extremely rare. I think `inner_product` might be a good follow up but not necessary for this PR..